### PR TITLE
Add --airflowVersion flag to $ astro deployment create

### PR DESCRIPTION
**Pull request by andriisoldatenko**
### Description

> Describe the purpose of this pull request.

#### 🎟 Issue(s)

Resolves https://github.com/astronomer/issues/issues/1741
Resolves https://github.com/astronomer/issues/issues/1806
🎉 🎉  bonus features Added column AIRFLOW VERSION to deployment create/list/update:

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
practical tests:
```
astro deployment create new-deployment-1-10-10-airflow-k8s --executor=celery --airflowVersion=1.10.10
 NAME                                   DEPLOYMENT NAME       ASTRO      DEPLOYMENT ID                 TAG
 new-deployment-1-10-10-airflow-k8s     new-velocity-8501     0.17.0     ckguogf6x0685ewxtebr4v04x     %!s(MISSING)

 Successfully created deployment with Celery executor. Deployment can be accessed at the following URLs

 Airflow Dashboard: https://deployments.local.astronomer.io/new-velocity-8501/airflow
 Flower Dashboard: https://deployments.local.astronomer.io/new-velocity-8501/flower
```

**Comment by andriisoldatenko** [comment]
@adam2k PTAL.

**Comment by andriisoldatenko** [comment]
practical tests for :

```
astro deployment airflow upgrade --deployment-id=ckguogf6x0685ewxtebr4v04x
#     AIRFLOW VERSION
1     1.10.7
2     1.10.10
3     1.10.12

> 3
 NAME                                   DEPLOYMENT NAME       ASTRO       DEPLOYMENT ID                 AIRFLOW VERSION
 new-deployment-1-10-10-airflow-k8s     new-velocity-8501     v0.17.0     ckguogf6x0685ewxtebr4v04x     1.10.12

The upgrade from Airflow 1.10.10 to 1.10.12 has been started.To complete this process, replace the image referenced in your Dockerfile and deploy to Astronomer.
To cancel, run:
 $ astro deployment airflow upgrade --cancel

$ astro deployment airflow upgrade --cancel --deployment-id=ckguogf6x0685ewxtebr4v04x

Airflow upgrade process has been successfully canceled. You are using right now 1.10.10
$ astro deployment airflow upgrade --cancel --deployment-id=ckguogf6x0685ewxtebr4v04x

Nothing to cancel. You are currently using airflow version 1.10.10 and desired airflow version is 1.10.10
```

**Comment by andriisoldatenko** [comment]
@paolaperaza please check wording 🙏

**Comment by andriisoldatenko** [comment]
bonus feature:
```
astro deployment list
 NAME                                            DEPLOYMENT NAME                 ASTRO       DEPLOYMENT ID                 TAG             AIRFLOW VERSION
 new-deployment-1-10-10-airflow-k8s-2            elementary-rotation-5522        v0.17.0     ckgwdq8cs037169xtbt2rtu15     deploy-2020     1.10.12
 new-deployment-1-10-10-airflow-k8s              new-velocity-8501               v0.17.0     ckguogf6x0685ewxtebr4v04x     deploy-2020     1.10.10
 asdnew-deployment-1-10-10-airflow-k8s-11212     quasarian-wave-5291             v0.17.0     ckgupaxk30037tbxtrmpzklj2     deploy-2020     1.10.10
 asdasd!!!!!!                                    telescopic-probe-7702           v0.17.0     ckgwdhd3k1050e2xtj86txbjp     deploy-2020     1.10.7-7
 asdasd                                          positional-magnitude-3427       v0.17.0     ckgup0f770452vcxtdfjmdu8f     deploy-2020     1.10.7-15
 aaa                                             explosive-spectroscope-0668     v0.17.0     ckguov22y0210bmxtxw02bsd6     deploy-2020     1.10.7-15
 1.10.12                                         planetic-conjunction-9372       v0.17.0     ckgwbpuen0706e2xtg0rel8m5     deploy-2020     1.10.7-7
 1.10.10                                         geostationary-parsec-7581       v0.17.0     ckgwboxdy0414e2xt7gv8rgvq     deploy-2020     1.10.7-7
```

File: deployment/deployment_test.go [hunk 1]
```diff
@@ -326,3 +326,65 @@ func TestUpdateError(t *testing.T) {
...
+			Body:       ioutil.NopCloser(bytes.NewBufferString(okResponse)),
+			Header:     make(http.Header),
+		}
+	})
+	api := houston.NewHoustonClient(client)
+	deploymentId := "ckbv818oa00r107606ywhoqtw"
+
+	deployment := getDeployment(deploymentId, api)
+	assert.Equal(t, deployment, &houston.Deployment{Id: "ckggzqj5f4157qtc9lescmehm", Label: "test", AirflowVersion: "1.10.5", DesiredAirflowVersion: "1.10.10"})
+}
```
Comment (adam2k) [inline deployment/deployment_test.go:390]: Can you add EOF line?

File: deployment/deployment_test.go [hunk 2]
```diff
@@ -326,3 +326,65 @@ func TestUpdateError(t *testing.T) {
 	err := Update(id, role, deploymentConfig, api, buf)
 	assert.EqualError(t, err, "API error (500): Internal Server Error")
 }
+
+func TestAirflowUpgrade(t *testing.T) {
```
Comment (adam2k) [inline deployment/deployment_test.go:330]: Can you add the error tests for both of the tests in this file?

**Review by adam2k — COMMENTED** [review]
This looks good to me.  It needs some error case tests, but other than that I think it's good to ship.

File: deployment/deployment.go [hunk 3]
```diff
@@ -183,9 +189,135 @@ func Update(id, cloudRole string, args map[string]string, client *houston.Client
...
+		Variables: vars,
+	}
+
+	r, err := req.DoWithClient(client)
+	if err != nil {
+		return nil
+	}
+
+	return &r.Data.GetDeployment
+}
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply deployment/deployment.go]: fixed.

File: deployment/deployment_test.go (hunk 1 above)
Comment (andriisoldatenko) [reply deployment/deployment_test.go:429]: fixed.

File: deployment/deployment_test.go (hunk 2 above)
Comment (andriisoldatenko) [reply deployment/deployment_test.go:330]: fixed.

File: deployment/deployment_test.go [hunk 4]
```diff
@@ -326,3 +327,153 @@ func TestUpdateError(t *testing.T) {
...
+			Body:       ioutil.NopCloser(bytes.NewBufferString(response)),
+			Header:     make(http.Header),
+		}
+	})
+	api := houston.NewHoustonClient(client)
+	deploymentId := "ckbv818oa00r107606ywhoqtw"
+
+	_, err := getDeployment(deploymentId, api)
+	assert.Error(t, err, "tes")
+}
```
Comment (adam2k) [inline deployment/deployment_test.go:479]: EOF is missing here too.

**Review by adam2k — APPROVED** [review]
This looks good to me.  There's one EOF line missing on the test file.

**Comment by adam2k** [comment]
Actually, it also looks like `getAirflowVersionSelection` still isn't covered because the branch is never reached where `desiredAirflowVersion == "" ` in the `AirflowUpgrade` function.

**Comment by andriisoldatenko** [comment]
> Actually, it also looks like `getAirflowVersionSelection` still isn't covered because the branch is never reached where `desiredAirflowVersion == "" ` in the `AirflowUpgrade` function.

please check latest commit

**Comment by adam2k** [comment]
> > Actually, it also looks like `getAirflowVersionSelection` still isn't covered because the branch is never reached where `desiredAirflowVersion == "" ` in the `AirflowUpgrade` function.
> 
> @adam2k please check latest commit [1450a5b](https://github.com/astronomer/astro-cli/commit/1450a5b874302aa54b25108bcf21947958e62017)

Yeah, that looks good.  I think codecov was cached on the previous commit for me.

File: deployment/deployment.go [hunk 5]
```diff
@@ -183,9 +189,141 @@ func Update(id, cloudRole string, args map[string]string, client *houston.Client
...
+
+	d := r.Data.UpdateDeploymentAirflow
+	tab := &printutil.Table{
+		Padding:        []int{30, 30, 10, 50, 10},
+		DynamicPadding: true,
+		Header:         []string{"NAME", "DEPLOYMENT NAME", "ASTRO", "DEPLOYMENT ID", "AIRFLOW VERSION"},
+	}
+	tab.AddRow([]string{d.Label, d.ReleaseName, "v" + d.Version, d.Id, d.AirflowVersion}, false)
+
+	tab.SuccessMsg = fmt.Sprintf("\nThe upgrade from Airflow %s to %s has been started.", d.AirflowVersion, d.DesiredAirflowVersion) +
```
Comment (andriisoldatenko) [inline deployment/deployment.go:229]: @paolaperaza PTAL

File: deployment/deployment.go [hunk 6]
```diff
@@ -183,9 +189,141 @@ func Update(id, cloudRole string, args map[string]string, client *houston.Client
...
+			Query:     houston.UpdateDeploymentAirflowRequest,
+			Variables: vars,
+		}
+
+		_, err := req.DoWithClient(client)
+		if err != nil {
+			return err
+		}
+
+		text := "\nAirflow upgrade process has been successfully canceled. You are using right now %s\n"
```
Comment (andriisoldatenko) [inline deployment/deployment.go:258]: @paolaperaza ^^

File: deployment/deployment.go [hunk 7]
```diff
@@ -183,9 +189,141 @@ func Update(id, cloudRole string, args map[string]string, client *houston.Client
...
+		fmt.Fprintf(out, text, deployment.AirflowVersion)
+		return nil
+	}
+
+	text := "\nNothing to cancel. You are currently using airflow version %s and desired airflow version is %s\n"
```
Comment (andriisoldatenko) [inline deployment/deployment.go:263]: @paolaperaza

File: cmd/deployment.go [hunk 8]
```diff
@@ -32,6 +35,9 @@ var (
 
 # Create new deployment with Kubernetes executor.
   $ astro deployment create new-deployment-name-k8s --executor=k8s
+
+# Create new deployment with Kubernetes executor.
+  $ astro deployment create new-deployment-1-10-10-airflow-k8s --executor=k8s --airflowVersion=1.10.10
```
Comment (paolaperaza) [inline cmd/deployment.go:40]: @andriisoldatenko I'm thinking out loud but I almost feel like this "deployment name" might confuse the user and make them think that what they name their deployment affects whether or not these flags are successfully passed.

Should we just do this as the example:

`$ astro deployment create my-new-deployment --executor=k8s --airflowVersion=1.10.10`

WDYT?

File: cmd/deployment_test.go [hunk 9]
```diff
@@ -251,3 +251,60 @@ func TestDeploymentUserUpdateCommand(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Contains(t, output, expectedOut)
 }
+
+func TestDeploymentAirflowUpgradeCommand(t *testing.T) {
+	testUtil.InitTestConfig()
+	expectedOut := `The upgrade from Airflow 1.10.5 to 1.10.10 has been started.To complete this process, replace the image referenced in your Dockerfile and deploy to Astronomer.`
```
Comment (paolaperaza) [inline cmd/deployment_test.go:257]: - Add space between `d.` and `To`
- Should we say, `To complete this process, add a 1.10.10 image to your Dockerfile and deploy to Astronomer`?

(Assuming we have "desired version" here - if not, then ^this looks good).

File: cmd/deployment_test.go [hunk 10]
```diff
@@ -251,3 +251,60 @@ func TestDeploymentUserUpdateCommand(t *testing.T) {
...
+
+	_, output, err := executeCommandC(api, "deployment", "airflow", "upgrade", "--deployment-id=ckggvxkw112212kc9ebv8vu6p", "--desired-airflow-version=1.10.10")
+	assert.NoError(t, err)
+	assert.Contains(t, output, expectedOut)
+}
+
+
+func TestDeploymentAirflowUpgradeCancelCommand(t *testing.T) {
+	testUtil.InitTestConfig()
+	expectedOut := `Airflow upgrade process has been successfully canceled. You are using right now 1.10.5`
```
Comment (paolaperaza) [inline cmd/deployment_test.go:286]:
```suggestion
	expectedOut := `Airflow upgrade process has been successfully canceled. You are still running Airflow 1.10.5. `
```

File: deployment/deployment.go (hunk 6 above)
Comment (paolaperaza) [reply deployment/deployment.go:258]:
```suggestion
		text := "\nAirflow upgrade process has been successfully canceled. Your Deployment was not interrupted and you are still running Airflow %s\n."
```

File: deployment/deployment.go (hunk 7 above)
Comment (paolaperaza) [reply deployment/deployment.go:263]:
```suggestion
	text := "\nNothing to cancel. You are currently running Airflow %s and you have not indicated that you want to upgrade."
```

File: deployment/deployment.go [hunk 11]
```diff
@@ -183,9 +189,141 @@ func Update(id, cloudRole string, args map[string]string, client *houston.Client
...
+	d := r.Data.UpdateDeploymentAirflow
+	tab := &printutil.Table{
+		Padding:        []int{30, 30, 10, 50, 10},
+		DynamicPadding: true,
+		Header:         []string{"NAME", "DEPLOYMENT NAME", "ASTRO", "DEPLOYMENT ID", "AIRFLOW VERSION"},
+	}
+	tab.AddRow([]string{d.Label, d.ReleaseName, "v" + d.Version, d.Id, d.AirflowVersion}, false)
+
+	tab.SuccessMsg = fmt.Sprintf("\nThe upgrade from Airflow %s to %s has been started.", d.AirflowVersion, d.DesiredAirflowVersion) +
+		"To complete this process, replace the image referenced in your Dockerfile and deploy to Astronomer.\n" +
```
Comment (paolaperaza) [inline deployment/deployment.go:230]: @andriisoldatenko Whoops, messaged this above but how about:

```suggestion
		"To complete this process, add an Airflow %s image to your Dockerfile and deploy to Astronomer.\n" +
```

`Airflow %s` = "Desired Airflow Version"

**Review by paolaperaza — COMMENTED** [review]
@andriisoldatenko I just left comments on copy! Let me know what you think.

**Comment by andriisoldatenko** [comment]
@paolaperaza fixed! PTAL

File: deployment/deployment.go (hunk 6 above)
Comment (andriisoldatenko) [reply deployment/deployment.go]: fixed.

File: deployment/deployment.go (hunk 7 above)
Comment (andriisoldatenko) [reply deployment/deployment.go]: fixed.

File: cmd/deployment.go (hunk 8 above)
Comment (andriisoldatenko) [reply cmd/deployment.go]: fixed in 4ece5bd962c8dd565dbf0215ef45af1f87576765.

**Pull request merged**